### PR TITLE
Host video sync

### PR DESCRIPTION
**Pull request by Erol444**
added sync functionality when sending video from the host

### CODE REVIEW

File: depthai_demo.py [hunk 1]
```diff
@@ -562,6 +568,9 @@ def create_system_logger(self):
 
     if conf.useCamera:
         pv.create_queues(device)
+    elif conf.args.sync:
+        conf.args.show = ["host"]
```
Comment (VanDavv) [inline depthai_demo.py:572]: I would move this to adjustPreviewToOptions - https://github.com/luxonis/depthai/blob/1f2244c75a660272bdef15f1865a4fda8b3fcd6a/depthai_helpers/config_manager.py#L184

File: depthai_demo.py [hunk 2]
```diff
@@ -485,7 +491,7 @@ def create_nn(self):
                 nn.passthrough.link(self.nodes.xout_nn_input.input)
 
             if conf.args.sync:
-                if self.nn_manager.source == "color" and hasattr(self.nodes, "xout_rgb"):
+                if self.nn_manager.source in ["color", "host"] and hasattr(self.nodes, "xout_rgb"):
```
Comment (VanDavv) [inline depthai_demo.py:494]: I wouldn't like to mix xout_rgb to send the host frames. Let's add another if condition and check for `xout_host` (which we create prior)

File: depthai_demo.py [hunk 3]
```diff
@@ -259,6 +260,11 @@ def create_nn_pipeline(self, p, nodes):
             xin.setStreamName(self.input_name)
             xin.out.link(nn.input)
             setattr(nodes, self.input_name, xout)
+            # Send the video frame back to the host
+            if conf.args.sync:
+                nodes.xout_rgb = p.createXLinkOut()
```
Comment (VanDavv) [inline depthai_demo.py:265]:
```suggestion
                nodes.xout_host = p.createXLinkOut()
```

File: depthai_demo.py [hunk 4]
```diff
@@ -259,6 +260,11 @@ def create_nn_pipeline(self, p, nodes):
...
+                nodes.xout_rgb.setStreamName("host")
```
Comment (VanDavv) [inline depthai_demo.py:266]: Let's add a `host` to Previews - https://github.com/luxonis/depthai/blob/1f2244c75a660272bdef15f1865a4fda8b3fcd6a/depthai_demo.py#L59 and use `Previews.host.name` as a stream name

File: depthai_demo.py [hunk 5]
```diff
@@ -600,6 +609,10 @@ def create_system_logger(self):
             # if high quality, send original frames
             if not conf.useHQ:
                 host_frame = scaled_frame
+            # If sync, wait for the NN passthrough frame from the device
+            if conf.args.sync:
+                host_frame_in = np.array(pv.output_queues[0].get().getData()).reshape((3, in_h, in_w)).transpose(1, 2, 0).copy()
```
Comment (VanDavv) [inline depthai_demo.py:614]: this should be updated in `prepare_frames` of PreviewManager

**Review by VanDavv — CHANGES_REQUESTED** [review]

**Comment by VanDavv** [comment]
Updated, testable using

```
depthai_demo.py -sync -vid https://www.youtube.com/watch?v=Y1jTEyb3wiI
```

**Review by VanDavv — APPROVED** [review]

File: depthai_demo.py [hunk 6]
```diff
@@ -606,17 +617,18 @@ def create_system_logger(self):
...
-                        print("Received NN packet: ", to_tensor_result(packet))
-                    except Exception as ex:
-                        print("Received NN packet: <Preview unabailable: {}>".format(ex))
-                fps.tick('nn')
+                detections = in_nn.detections
+            elif nn_manager.output_format is None:
+                try:
+                    print("Received NN packet: ", to_tensor_result(in_nn))
+                except Exception as ex:
+                    print("Received NN packet: <Preview unabailable: {}>".format(ex))
```
Comment (cafemoloko) [inline depthai_demo.py:630]: @VanDavv, typo in word _unavailable_

**Pull request merged**